packages/client/src/features/change-bounds/resize-tool.ts

```diff
@@ -0,0 +1,157 @@
+            if (!this.isEditMode && matchesKeystroke(event, 'KeyR', 'alt')) {
+                this.isEditMode = true;
+            }
+
+            if (this.isEditMode) {
+                const selectedElements = this.getSelectedElements(element.root);
+                if (event.key === '+') {
+                    for (const elem of selectedElements) {
+                        const action = this.handleResizeElement(elem, this.grid.x, this.grid.y);
+                        if (action) {
+                            actions.push(action);
+                        }
+                    }
+                } else if (matchesKeystroke(event, 'Minus')) {
+                    for (const elem of selectedElements) {
+                        const action = this.handleResizeElement(elem, -this.grid.x, -this.grid.y);
+                        if (action) {
+                            actions.push(action);
+                        }
+                    }
+                } else if (matchesKeystroke(event, 'Digit0', 'ctrl')) {
+                    for (const elem of selectedElements) {
+                        elem.bounds = { x: elem.bounds.x, y: elem.bounds.y, width: minWidth(elem), height: minHeight(elem) };
+                        const action = this.handleResizeElement(elem, 0, 0);
+                        if (action) {
+                            actions.push(action);
+                        }
+                    }
+                }
+            }
+        }
+
+        return actions;
+    }
+
+    protected handleResizeElement(element: SModelElement & BoundsAware, deltaWidth: number, deltaHeight: number): Action | undefined {
+        const x = element.bounds.x;
+        const y = element.bounds.y;
+        const width = element.bounds.width + deltaWidth;
+        const height = element.bounds.height + deltaHeight;
+        const newPosition = { x, y };
+        const newSize = { width, height };
+
+        const resizeElement = { id: element.id, bounds: { x, y, width, height } } as SModelElement & SParentElement & Resizable;
+
+        if (this.isValidBoundChange(resizeElement, newPosition, newSize)) {
+            return ChangeBoundsOperation.create([toElementAndBounds(resizeElement)]);
+        }
+
+        return undefined;
+    }
+
+    protected isValidBoundChange(element: SModelElement & BoundsAware, newPosition: Point, newSize: Dimension): boolean {
+        return this.isValidSize(element, newSize) && this.isValidMove(element, newPosition);
+    }
+    protected isValidSize(element: SModelElement & BoundsAware, size: Dimension): boolean {
+        return isValidSize(element, size);
+    }
+
+    protected isValidMove(element: SModelElement & BoundsAware, newPosition: Point): boolean {
+        return isValidMove(element, newPosition, this.tool.movementRestrictor);
+    }
+
+    protected getSelectedElements(root: SModelRoot): (SModelElement & BoundsAware & Selectable)[] {
+        return Array.from(
```

Comment:
minor: If you were to inject the selection service in the tool, you could make this method a bit easier to read:
```
    protected getSelectedElements(): SelectableBoundsAware[] {
        const index = this.tool.selectionService.getModelRoot().index;
        const selectedElements = this.tool.selectionService.getSelectedElementIDs();
        return getElements(index, selectedElements, isSelectableAndBoundsAware);
    }
```